File: Backend/HoPla_API/Entities/Item.cs
﻿using System.ComponentModel.DataAnnotations;

namespace HoPla_API.Entities
{
    public class Item
    {
        [Key]
        public int Id { get; set; }

        [Required]
        [MaxLength(50)]
        public string? Name { get; set; }

        public ICollection<Reservation>? Reservations { get; set; }

        public string? Image { get; set; }

        public string? Qrcode { get; set; }

        public Item(string name, string image, string qrcode)
        {
            Name = name;
            Image = image;
            Qrcode = qrcode;
            Reservations = new List<Reservation>();
        }
    }

    public class ItemInputModel
    {
        [MaxLength(50)]
        public string Name { get; set; }

        public int HouseId { get; set; }

        public string? Image { get; set; }

        public string? Qrcode { get; set; }
    }
}

File: Frontend/lib/model/item.dart
import 'package:json_annotation/json_annotation.dart';

import 'reservation.dart';

part 'item.g.dart';

@JsonSerializable()
class Item {
  int id;
  String name;
  List<Reservation>? reservations;
  String image;
  String? qrcode;

  Item(
      {required this.id,
      required this.name,
      required this.reservations,
      required this.image,
      required this.qrcode});

  factory Item.fromJson(Map<String, dynamic> json) => _$ItemFromJson(json);

  Map<String, dynamic> toJson() => _$ItemToJson(this);
}
